File: src/main/java/org/normal/api/registry/Registry.java
package org.normal.api.registry;

import java.util.Map;

public interface Registry<K, T> {

    Map<K, T> getBackingMap();

    void register(K key, T object);

    void unregister(K key);

    void set(Map<K, T> elements);

    T get(K key);

    void clearEntries();

}

File: src/main/java/org/normal/api/kt/registry/Registry.kt
package org.normal.api.kt.registry

interface Registry<K, T> {

    fun getBackingMap(): Map<K, T>

    fun register(key: K, obj: T)

    fun unregister(key: K)

    fun set(elements: Map<K, T>)

    fun get(key: K): T?

    fun clearEntries()

}